app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt

```diff
@@ -74,6 +74,7 @@ class StateFragmentPresenter @Inject constructor(
   private lateinit var currentStateName: String
   private lateinit var binding: StateFragmentBinding
   private lateinit var recyclerViewAdapter: RecyclerView.Adapter<*>
+  private val hintBulbVibisilityStack: MutableList<Boolean> = mutableListOf()
```

Comment:
I wonder if we can simplify this. Do we ever need to show hints or a solution for previous states? If not, rather than tracking each state's hint visibility status, all we need to do is only track the current state's.

Reply:
Done